lib/environment/auth.js

```diff
@@ -1,23 +1,29 @@
 /* @flow */
 
 import { sendMessage } from '../../browser';
+import { CACHED_GOOGLE_EMAIL_ADDRESS } from '../constants/localStorage';
 import { Permissions } from './';
 
 export async function launchAuthFlow(
-	{ domain, clientId, scope = '' }: {| domain: string, clientId: string, scope?: string |},
+	{ domain, clientId, scope = '', loginHint = '' }: {| domain: string, clientId: string, scope?: string, loginHint?: string |},
 	warnUserInteraction: (message: string) => Promise<void>,
 ): Promise<string> {
 	let responseUrl;
 	try {
-		responseUrl = await sendMessage('authFlow', { domain, clientId, scope, interactive: false });
+		responseUrl = await sendMessage('authFlow', { domain, clientId, scope, loginHint, interactive: false });
 	} catch (e) {
 		console.error('Noninteractive auth failed:', e);
 
+		// Invalidate google email address
```

Comment:
It would be cleaner doing in `GoogleDrive.js`. Perhaps inside the `warnUserInteraction` callback?